icu4j/main/classes/core/src/com/ibm/icu/number/Precision.java

```diff
@@ -552,15 +551,29 @@ int chooseMultiplierAndApply(DecimalQuantity input, MultiplierProducer producer)
     // INTERNALS //
     ///////////////
 
-    static class BogusRounder extends Precision {
+    public static class BogusRounder extends Precision {
         @Override
         public void apply(DecimalQuantity value) {
-            // No-op
+            throw new AssertionError("BogusRounder must not be applied");
         }
 
         @Override
         BogusRounder createCopy() {
-            return new BogusRounder();
+            BogusRounder copy = new BogusRounder();
+            copy.mathContext = mathContext;
+            return copy;
+        }
+
+        /**
+         * Copies the BogusRounder's MathContext into precision.
+         *
+         * @internal
+         * @deprecated This API is ICU internal only.
+         */
+        @Deprecated
+        public Precision into(Precision precision) {
+            precision.mathContext = mathContext;
```

Comment:
You should copy the precision first.  It might be a singleton.

```
Precision copy = precision.createCopy();
copy.mathContext = this.mathContext;
return copy;
```